packages/peregrine/lib/Router/resolveUnknownRoute.js

```diff
@@ -5,7 +5,14 @@ import BrowserPersistence from '../util/simplePersistence';
  * @param {{ route: string, apiBase: string, __tmp_webpack_public_path__: string}} opts
  */
 const persistence = new BrowserPersistence();
-const routeCacheKey = 'urlResolver';
+
+/** Get and Set current store view */
+if (typeof persistence.getItem('store_view') === 'undefined' || typeof persistence.getItem('locale') === 'undefined') {
+    persistence.setItem('store_view', DEFAULT_STORE_VIEW.code);
+    persistence.setItem('locale', DEFAULT_STORE_VIEW.locale);
+}
+
+const routeCacheKey = 'urlResolver_' + persistence.getItem('store_view');
```

Comment:
So I think the recommendation from you / the team was to steer away from ContextProviders. I did review it initially but did not in the end go with it because the context providers are added after the first call we need to the localisation service.